geonode/security/utils.py

```diff
@@ -238,13 +231,15 @@ def skip_registered_members_common_group(user_group):
     return False
 
 
-def get_user_groups(owner):
+def get_user_groups(owner, group=None):
     """
     Returns all the groups belonging to the "owner"
     """
     user_groups = Group.objects.filter(
         name__in=owner.groupmember_set.all().values_list("group__slug", flat=True))
```

Comment:
if the `values_list` is called, the `.all()` is not needed, it will slow down the performances, plus the iterator can be called directly here. The `list(set())` in the return will covert it anyway.
In the case of appending the group, the group can be converted as a generator by using 
`(group.group if hasattr(group, 'group') else group)`
Then the two generators can be joined using the chain
```python
from itertools import chain
groups = []
return list(set(chain(user_groups, [])))
```

Reply:
Done